diff -Naur a/art/dex2oat/dex2oat.cc b/art/dex2oat/dex2oat.cc
--- a/art/dex2oat/dex2oat.cc
+++ b/art/dex2oat/dex2oat.cc
@@ -451,10 +451,10 @@ class WatchDog {
   // desktop. Debug builds are slower so they have larger timeouts.
   static constexpr int64_t kWatchdogSlowdownFactor = kIsDebugBuild ? 5U : 1U;
 
-  // 9.5 minutes scaled by kSlowdownFactor. This is slightly smaller than the Package Manager
-  // watchdog (PackageManagerService.WATCHDOG_TIMEOUT, 10 minutes), so that dex2oat will abort
+  // 59.5 minutes scaled by kSlowdownFactor. This is slightly smaller than the Package Manager
+  // watchdog (PackageManagerService.WATCHDOG_TIMEOUT, 60 minutes), so that dex2oat will abort
   // itself before that watchdog would take down the system server.
-  static constexpr int64_t kWatchDogTimeoutSeconds = kWatchdogSlowdownFactor * (9 * 60 + 30);
+  static constexpr int64_t kWatchDogTimeoutSeconds = kWatchdogSlowdownFactor * (59 * 60 + 30);
 
   static constexpr int64_t kDefaultWatchdogTimeoutInMS =
       kWatchdogVerifyMultiplier * kWatchDogTimeoutSeconds * 1000;
